action/completion/generate.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package completion
+
+import (
+	"fmt"
+
+	"github.com/go-vela/cli/internal/output"
+)
+
+// Generate produces a script used to enable
+// automatic completion for a Unix shell
+// based off the provided configuration.
+func (c *Config) Generate() error {
```

Comment:
This is where we add the new `Generate()` function.

The purpose of this function is to generate a shell auto completion script that can be sourced into the current terminal session.